File: Solutions/1380- Lucky Numbers in a Matrix/Solution.py
class Solution:
    def luckyNumbers (self, matrix: List[List[int]]) -> List[int]:
        size = len(matrix[0])
        count = 0
        my_set = set()
        answer = list()

        for item in matrix:
            my_set.add(min(item))

        max_number = -1
        while count < size:
            for item in matrix:
                if item[count] > max_number:
                    max_number = item[count]
            if max_number in my_set:
                answer.append(max_number)
            count += 1
            max_number = -1

        return answer

File: Solutions/1380- Lucky Numbers in a Matrix/Solution.java
class Solution {
    public List<Integer> luckyNumbers (int[][] matrix) {
        int size = matrix[0].length;
        int count = 0;
        Set<Integer> mySet = new HashSet<>();
        List<Integer> answer = new ArrayList<>();

        for (int[] row : matrix) {
            int minElement = Integer.MAX_VALUE;
            for (int num : row) {
                if (num < minElement) {
                    minElement = num;
                }
            }
            mySet.add(minElement);
        }

        int maxNumber = -1;
        while (count < size) {
            for (int[] row : matrix) {
                if (row[count] > maxNumber) {
                    maxNumber = row[count];
                }
            }
            if (mySet.contains(maxNumber)) {
                answer.add(maxNumber);
            }
            count++;
            maxNumber = -1;
        }

        return answer;
    
    }
}